Hades: cleanup all database operations for upcoming MongoDB migration

akhilnarang opened the pull request.

akhilnarang committed:
Hades: cleanup all db operations

Move almost everything to db_utils, besides password related transactions (for now)

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

ksdfg reviewed (approved):
Seems fine, can't see anything wrong in here

akhilnarang committed:
Hades: remove the last usages of the `db` object from `__init__.py`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: general cleanup, use `id_` instead of `id`

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: utils: update docstrings for parameter name change

Update get_id() except statement

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: db_utils: fix get_user()

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: cleanup utils, add some TODOs

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

ksdfg committed:
update black dep

akhilnarang committed:
Hades: `/login`: update error message when user doesn't exist

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: fix `get_user()` call

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Hades: logout current user on `/register`

Although it does work fine, seems meh, since the old user stays logged
in after registration of the new user

Now we logout whenever `/register` is opened (if someone is logged in),
and we login to the newly registered user after registration is done

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
